torch/testing/_internal/common_device_type.py

```diff
@@ -377,6 +377,14 @@ def instantiated_test(self, param_kwargs=param_kwargs):
                 except RuntimeError as rte:
                     # check if rte should stop entire test suite.
                     self._stop_test_suite = self._should_stop_test_suite()
+                    # Check if test has been decorated with `@expectedFailure`
+                    # Based on https://github.com/python/cpython/blob/ffa505b580464d9d90c29e69bd4db8c52275280a/Lib/unittest/case.py#L164
+                    # Suppress the decorator in that case
+                    # As it might hide followup failures
+                    if getattr(test, "__unittest_expecting_failure__", False) and self._stop_test_suite:
+                        from warnings import warn
+                        warn("Suppressing expectedFailur decorator which resulted in fatal error")
+                        self.__unittest_expecting_failure__ = False
```

Comment:
haha clever let's see if it works

Reply:
Hm, so CI would still show green :-( But at least logs would be better. Never xfail a test that causes segfault.

Reply:
I think suppressing exception to make expectedFail test to fail would make sense here